pilot_experiment/functions/calculations.py

```diff
@@ -0,0 +1,245 @@
+                                )
+
+    # Accessory gene stats
+    acc_stats_df = pd.DataFrame(data={'ref_gene':acc_gene_ids,
+                                    'num_coexpressed_genes':num_coexpressed_genes[acc_gene_ids],
+                                    'num_coexpressed_core': [num_coexpressed_core_genes[a] for a in acc_gene_ids],
+                                    'num_coexpressed_acc': [num_coexpressed_acc_genes[a] for a in acc_gene_ids],
+                                    'percent_coexpressed_core': [coexpressed_core_prop[a] for a in acc_gene_ids],
+                                    'percent_coexpressed_acc': [coexpressed_acc_prop[a] for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_genes':[num_non_cooperonic_coexpressed_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_core': [num_non_cooperonic_coexpressed_core_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'num_non_cooperonic_coexpressed_acc': [num_non_cooperonic_coexpressed_acc_genes[a] 
+                                                                            for a in acc_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_core': [non_cooperonic_coexpressed_core_prop[a] 
+                                                                                for a in acc_gene_ids],
+                                    'percent_non_cooperonic_coexpressed_acc': [non_cooperonic_coexpressed_acc_prop[a] 
+                                                                                for a in acc_gene_ids]
+                                    }
+                                )
+    return core_stats_df, acc_stats_df
+            
+
+def map_gene_to_operon(operon_annotation_file,
+all_gene_ids):
+    """
+    Returns mapping between gene id and list of co-operonic genes
+
+    Arguments
+    ----------
+    operon_annotation_file: str
+        File containing list of genes per operon
+
+    all_gene_ids: list
+        List of all gene ids to search for co-operonic genes 
+    """
+
+    # Read operon data
+    # Manually had to set names to be the max size operon
+    operon_data = pd.read_csv(
+        operon_annotation_file,
+        header=None,
+        sep='\t',
+        names=range(15)
+    )
+    # Associate operons with reference gene
+    name_cooperonic_genes = {}
+
+    for gene_id in all_gene_ids:
+        # Get operons containing reference gene_id   
+        # Search for gene_id in each operon
+        operon_search = operon_data.where(operon_data == gene_id).dropna(how='all').dropna(axis=1)
+        
+        if operon_search.empty:
+            name_cooperonic_genes[gene_id] = []
+        else:
+            row_id = operon_search.index[0]
```

Comment:
Good point. This is setup with way because I knew that in this case there was a 1-1 mapping. I will definitely add a comment. I am not sure if a gene will belong to multiple operons.